Shell/main.cpp

```diff
@@ -147,6 +147,43 @@ static int sh_unset(int argc, char** argv)
     return 0;
 }
 
+static String expanded_tilde_expression_from(char* arg)
+{
+    int len = strlen(arg);
+    ASSERT(len);
+
+    StringBuilder login_name;
+    int first_slash_index = len;
+    for (int i = 1; i < len; ++i) {
+        if (arg[i] == '/') {
+            first_slash_index = i;
+            break;
+        }
+        login_name.append(arg[i]);
+    }
```

Comment:
That is a weird way to spell `strchr()` or `split_view()[0]` 😀

Reply:
Ah thanks!

Reply:
I tried both of those different ways, and honestly I still prefer doing it this way as it makes building the path really easy

Reply:
Actually, nevermind, just learnt about `strchrnul`

Reply:
Yeah tried that aswell. Still can't get it very readable for some reason. I must be doing something wrong to be honest.